jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaInfConfiguration.java

```diff
@@ -293,7 +293,10 @@ protected void scanJars(WebAppContext context) throws Exception
         //only look for fragments if web.xml is not metadata complete, or it version 3.0 or greater
         List<String> scanTypes = new ArrayList<>(__allScanTypes);
         if (context.getMetaData().isMetaDataComplete() || (context.getServletContext().getEffectiveMajorVersion() < 3) && !context.isConfigurationDiscovered())
+        {
             scanTypes.remove(METAINF_FRAGMENTS);
+            scanTypes.remove(METAINF_RESOURCES);
```

Comment:
The spec definitely says we should not look at `META-INF/web-fragment.xml`, however I don't see anywhere where it says that we shouldn't look for `META-INF/resources` or `META-INF tlds`. Can you point to the text that would support this change?

Reply:
TLDs are not impacted by the above change, so lets ignore that part of your question.

`META-INF/web-fragment.xml` and `META-INF/resources/` are both introduced in Servlet 3.0 - didn't exist prior, only reason.
In fact, `META-INF/resources/` didn't exist until _after_ the Servlet 3.0 public draft (that's how late it was added) - it only showed up in the Servlet 3.0 final spec.

Reply:
According to the spec `metadata-complete` means that everything is declared in `web.xml` (or in a `web-fragment.xml`) and annotations are not to be processed, except for those that have no equivalent in `web.xml`. See section 8.1 pg 71. Static resources have never been declared in `web.xml`, thus `metadata-complete` should have no impact on them.

Reply:
Potentially, maybe we should be excluding static resources from fragment jars that have been excluded by an ordering in `web.xml/web-fragment.xml`. Currently I think they will be added to the context base resource regardless of if they're excluded or not. We do the addition of `META-INF/resources` in `MetaInfConfiguration` before we have parsed `web.xml` and the `web-fragment.xmls` to find out whether there are any excluded fragments.  Of course, you can always use a jetty `org.eclipse.jetty.server.webapp.WebInfIncludeJarPattern` to exclude those jars from being examined and thus prevent their resources from being added to the context base resource.

Again, unfortunately the spec doesn't say anything about exclusion of static resources (or `tlds` for that matter) from jars that have been excluded by an ordering - it just refers to annotation processing and `web-fragment.xml`.

Reply:
Went ahead and rolled this change back.